easybuild/framework/easyblock.py

```diff
@@ -914,7 +914,9 @@ def make_module_extend_modpath(self):
             modpath_exts = ActiveMNS().det_modpath_extensions(self.cfg)
             self.log.debug("Including module path extensions returned by module naming scheme: %s" % modpath_exts)
             full_path_modpath_extensions = [os.path.join(top_modpath, mod_path_suffix, ext) for ext in modpath_exts]
-            # module path extensions must exist, otherwise loading this module file will fail
+            # module path extensions must exist since the generated module will run 'module use' on them;
+            # this is only really true for Tcl/C env mods, but we cannot make any assumptions on which modules tool
+            # will be used to consume the generated module (better safe than sorry)
             for modpath_extension in full_path_modpath_extensions:
```

Comment:
I'm not going to tackle these in here, this can be followed up by anyone in a follow-up PR.

Maybe something for @geimer, since he was reporting the empty directories that `eb` is putting in place, but which are not required with Lmod or `modulecmd.tcl`.